server/services/locationService.ts

```diff
@@ -1,15 +1,19 @@
-import { PrisonApiClient, WhereaboutsApiClient } from '../data'
+import { PrisonApiClient, WhereaboutsApiClient, LocationsInsidePrisonApiClient } from '../data'
 import { Agency, Location, OffenderCell } from '../data/prisonApiClient'
-import { LocationGroup, LocationPrefix } from '../data/whereaboutsApiClient'
+import { LocationGroup, LocationPrefix } from '../data/locationsInsidePrisonApiClient'
 
 export default class LocationService {
   constructor(
     private readonly prisonApiClient: PrisonApiClient,
     private readonly whereaboutsApiClient: WhereaboutsApiClient,
+    private readonly locationsInsidePrisonApiClient: LocationsInsidePrisonApiClient,
   ) {}
 
   async searchGroups(token: string, agencyId: string): Promise<LocationGroup[]> {
-    return await this.whereaboutsApiClient.searchGroups(token, agencyId)
+    const groups = await this.locationsInsidePrisonApiClient.searchGroups(token, agencyId)
+    return groups.map(group =>
+      group.children.length === 1 ? { name: group.name, key: group.key, children: [] } : group,
+    )
```

Comment:
this is reducing children having a single element to an empty array. This means that when a user selects a sub location dropdown that ordinarily has just one child, that child won't be displayed